client/client.go

```diff
@@ -446,6 +446,31 @@ func (c *Client) handleStartJobRequest(p *agentpb.StartJobRequest) error {
 			Socket:   j.MysqlBackup.Socket,
 		}
 		job = jobs.NewMySQLBackupJob(p.JobId, timeout, j.MysqlBackup.Name, cfg, locationConfig)
+
+	case *agentpb.StartJobRequest_MongodbBackup:
+		var locationConfig jobs.BackupLocationConfig
+		switch cfg := j.MongodbBackup.LocationConfig.(type) {
+		case *agentpb.StartJobRequest_MongoDBBackup_S3Config:
+			locationConfig.S3Config = &jobs.S3LocationConfig{
+				Endpoint:     cfg.S3Config.Endpoint,
+				AccessKey:    cfg.S3Config.AccessKey,
+				SecretKey:    cfg.S3Config.SecretKey,
+				BucketName:   cfg.S3Config.BucketName,
+				BucketRegion: cfg.S3Config.BucketRegion,
+			}
+		default:
+			return errors.Errorf("unknown location config: %T", j.MongodbBackup.LocationConfig)
+		}
+
+		cfg := jobs.DatabaseConfig{
```

Comment:
ConnectionConfig? Credentials?
DatabaseConfig feels like something which contains database configuration.

Reply:
Renamed to `DBConnConfig`